### PR TITLE
docker: add support for Alpine aarch64 image #2497

### PR DESCRIPTION
**Pull request by zdtsw**
- update Alpine play to install jdk16/17/18 from zulu which has support for aarch64
- add jenkins stage to build alpine image
- skip install test_tools packages

##### Checklist

- [x] commit message has one of the [standard prefixes](https://github.com/adoptium/infrastructure/blob/master/CONTRIBUTING.md#commit-messages)
- [x] [faq.md](https://github.com/adoptium/infrastructure/blob/master/FAQ.md) updated if appropriate
- [ ] other documentation is changed or added (if applicable)
- [ ] playbook changes run through [VPC](https://ci.adoptopenjdk.net/view/Tooling/job/VagrantPlaybookCheck/) or [QPC](https://ci.adoptopenjdk.net/view/Tooling/job/QEMUPlaybookCheck/) (if you have access)
- [ ] for inventory.yml changes, bastillion/nagios/jenkins updated accordingly

Ref: https://github.com/adoptium/infrastructure/issues/2497

### CODE REVIEW

**Comment by zdtsw** [comment]
several parts in this PR I am not 100% sure:
- skip [test_tools ](https://github.com/adoptium/infrastructure/blob/master/ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/vars/Alpine.yml#L46) which I do not think should be included in the final docker image release to end-user
- keep[build_tools](https://github.com/adoptium/infrastructure/blob/master/ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/vars/Alpine.yml#L7) but I am not sure if all these tools are needed. To only make ansible work, unzip tar should be enough
- other dockerfile (i.e for centos7) handle jenkins [user and group](https://github.com/adoptium/infrastructure/blob/master/ansible/docker/Dockerfile.CentOS7#L18) which I do not think is needed, that's why I did not add for alpine dockerfile. Should we cleanup this part later (in another PR)?

**Comment by sxa** [comment]
Please leave `test_tools` in for consistency with the other platforms in case we want to run some basic smoke tests inside the container.

**Comment by sxa** [comment]
I'm not quite sure why the username is handled differently, but for now I'm happy with leaving it the same as it does for Alpine/x64, and if we decide to change we can discuss more widely.

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/Alpine.yml [hunk 1]
```diff
@@ -24,94 +24,191 @@
     - not usr_lib_jvm_exists.stat.exists
   tags: build_tools
 
-- name: Check if zulu-8 is already installed in the target location
-  stat: path=/usr/lib/jvm/zulu8
-  register: zulu8_installed
-  tags: build_tools
+- name: Install JDK for aarch64
```
Comment (sxa) [inline ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/Alpine.yml:27]:
```suggestion
- name: Install JDK for x64
```

**Review by sxa — CHANGES_REQUESTED** [review]
One minor comment change, then I'm ok to approve and merge this :-D

**Review by sxa — APPROVED** [review]
Thanks - will merge once checks complete. We can probably prepare a PR in https://github.com/adoptium/ci-jenkins-pipelines to utilise this now - there are a few examples of adding extra builds in if you look at the PR history in that repository :-)

**Comment by zdtsw** [comment]
> Thanks - will merge once checks complete. We can probably prepare a PR in https://github.com/adoptium/ci-jenkins-pipelines to utilise this now - there are a few examples of adding extra builds in if you look at the PR history in that repository :-)

thanks, will do!

**Comment by zdtsw** [comment]
> Thanks - will merge once checks complete. We can probably prepare a PR in https://github.com/adoptium/ci-jenkins-pipelines to utilise this now - there are a few examples of adding extra builds in if you look at the PR history in that repository :-)

Sorry, but I thought https://ci.adoptopenjdk.net/job/centos7_docker_image_updater/ is triggered by any changes into "infrastructure" repo's default branch and load Jenkinsfile itself. 
anything  more in "ci-jenkins-pipeline" repo to enable build new docker image?

**Comment by sxa** [comment]
> > Thanks - will merge once checks complete. We can probably prepare a PR in https://github.com/adoptium/ci-jenkins-pipelines to utilise this now - there are a few examples of adding extra builds in if you look at the PR history in that repository :-)
> 
> Sorry, but I thought https://ci.adoptopenjdk.net/job/centos7_docker_image_updater/ is triggered by any changes into "infrastructure" repo's default branch and load Jenkinsfile itself. anything more in "ci-jenkins-pipeline" repo to enable build new docker image?

Yep that's correct - this PR should trigger the creation of the docker image, but we also need to tell jenkins to include Alpine/aarch64 bulids when it runs the regular pipelines, so we need to add it into the files in https://github.com/adoptium/ci-jenkins-pipelines/tree/master/pipelines/jobs/configurations for java to be built using those docker images.

**Pull request merged**